src/extensions/default/AutoUpdate/node/AutoUpdateDomain.js

```diff
@@ -167,6 +167,85 @@
         }
     }
 
+	/**
+     * Parse the Installer log and search for a error strings
+	 * one it finds the line which has any of error String
+     * it return that line and exit
+     */
+    function parseInstallerLog(filepath, searchstring, encoding, callback) {
+        var line = "";
+        var    searchFn = function (str) {
+            var arr = str.split('\n'),
+                lineNum,
+                pos;
+            for (lineNum = arr.length - 1; lineNum >= 0; lineNum--) {
+                var searchStrNum;
+                for (searchStrNum = 0; searchStrNum < searchstring.length; searchStrNum++) {
+                    pos = arr[lineNum].search(searchstring[searchStrNum]);
+                    if (pos !== -1) {
+                        line = arr[lineNum];
+                        break;
+                    }
+                }
+                if (pos !== -1) {
+                    break;
+                }
+            }
+            callback(line);
+        };
+
+        fs.readFile(filepath, {"encoding": encoding})
+            .then(function (str) {
+                return searchFn(str);
+            }).catch(function () {
+                callback("");
+            });
+    }
+
+    function checkInstallerStatus(searchParams) {
+        var installErrorStr = searchParams.installErrorStr,
+            bracketErrorStr = searchParams.bracketErrorStr,
+            updateDirectory = searchParams.updateDir,
+            encoding =        searchParams.encoding,
+            statusObj = {},
+            logFileAvailable = false;
+
+        if (!encoding) {
```

Comment:
you can remove this piece of code.
Change line 209 to -
``
encoding = searchParams.encoding || "utf8",
``